### PR TITLE
remove script/trigger/cantonFixture deps on scala-bindings

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
Basically we remove the dependency of different components on "//language-support/scala/bindings" by:

- replacing `com.daml.ledger.api.refinements.ApiTypes.Party` by `com.daml.lf.data.Ref.Party` 
- replacing `com.daml.ledger.api.refinements.ApiTypes.ApplicationId` by `Option[com.daml.lf.data.Ref.ApplicationId]` (here we use option as `ApiTypes.ApplicationId` allows empty string while `Ref.ApplicationId` does not).
- adding rounding logic for timestamp in `com.daml.lf.data.Time.Timestamp` and use it instead of the one from `com.daml.api.util.TimestampConversion`

Note we did not clean daml-sript export as it have never pass the alpha stage and will be dropped with the 3.x fork.

### CODE REVIEW

File: compiler/repl-service/server/src/main/scala/com/digitalasset/daml/lf/speedy/ReplServiceMain.scala [hunk 1]
```diff
@@ -51,7 +50,7 @@ object ReplServiceMain extends App {
       ledgerHost: Option[String],
       ledgerPort: Option[Int],
       accessTokenFile: Option[Path],
-      applicationId: Option[ApplicationId],
+      applicationId: Option[Option[Ref.ApplicationId]],
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply compiler/repl-service/server/src/main/scala/com/digitalasset/daml/lf/speedy/ReplServiceMain.scala:53]: This change is purely mechanical. I prefer not doing any "clever" stuff in this PR.
Old `ApiTypes.ApplicationId` allowed empty string while `Ref.Application` does not, so it is why I introduced the `Option` here. As far I can tell the Script were actively using empty `Application Id`, so you could end up here with an `Some("")`. I find that quite fishy and I suspect that something is wrong somewhere. But as do not understand all the consequence, I really prefer to not tackle that in this PR.

I will be gratefull, to double check that I propagate the Option everywhere. In particular some test use `equality` between values of  `Option[ApplicationId]` that is not type safe.

**Review by samuel-williams-da — COMMENTED** [review]
Looks good from a daml-script/compiler POV, in addition to Carl's comments

File: daml-script/test/BUILD.bazel [hunk 2]
```diff
@@ -174,8 +176,7 @@ da_scala_library(
         "//daml-lf/language",
         "//daml-lf/transaction",
         "//daml-script/runner:script-runner-lib",
-        "//language-support/scala/bindings",
-        "//language-support/scala/bindings-pekko",
+        "//ledger/ledger-api-client",
```
Comment (mziolekda) [inline daml-script/test/BUILD.bazel:179]: This is a sad panda :cry: :panda_face: in its own right. But let's fight that battle another day.

File: test-common/BUILD.bazel [hunk 3]
```diff
@@ -100,13 +100,6 @@ da_scala_dar_resources_library(
 [
     [
         [
-            dar_to_scala(
```
Comment (mziolekda) [inline test-common/BUILD.bazel:103]: nice

File: test-common/canton/it-lib/BUILD.bazel [hunk 4]
```diff
@@ -27,10 +28,11 @@ da_scala_library(
         "//bazel_tools/runfiles:scala_runfiles",
         "//daml-lf/archive:daml_lf_archive_reader",
         "//daml-lf/data",
-        "//language-support/scala/bindings",
-        "//language-support/scala/bindings-pekko",
+        "//ledger-api/grpc-definitions:ledger_api_proto_scala",
```
Comment (mziolekda) [inline test-common/canton/it-lib/BUILD.bazel:31]: that won't work when rebased on main

File: triggers/runner/BUILD.bazel [hunk 5]
```diff
@@ -28,16 +29,22 @@ da_scala_library(
         "//daml-lf/language",
         "//daml-lf/transaction",
         "//daml-script/converter",
-        "//language-support/scala/bindings",
-        "//language-support/scala/bindings-pekko",
+        "//ledger-api/grpc-definitions:ledger_api_proto_scala",
```
Comment (mziolekda) [inline triggers/runner/BUILD.bazel:32]: that won't work when rebased on main, same througout this PR, this needs to be `//canton:ledger_api_proto_scala`

Comment (remyhaemmerle-da) [reply triggers/runner/BUILD.bazel:32]: :+1:

File: daml-script/test/BUILD.bazel (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-script/test/BUILD.bazel:179]: Note that the script  and trigger runners are already using the ledger-api-client. 
If you want we drop it, you will have to give us an alternative, we are not really in mood to reimplement a ledger client from scratch.

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Time.scala [hunk 6]
```diff
@@ -112,57 +117,73 @@ object Time {
 
   object Timestamp {
 
+    // TODO: Daml3, lets change that for daml 3.
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Time.scala]: done.

File: daml-script/test/BUILD.bazel (hunk 2 above)
Comment (mziolekda) [reply daml-script/test/BUILD.bazel:179]: In fact, there is a LedgerClient in the ledger-api-core library of canton. It's a different discussion though whether a dependency on canton's entrails is desirable here.

**Review by samuel-williams-da — APPROVED** [review]
Stamp for just compiler/daml-script
Please do create a followup ticket/PR to address some of the more nuanced changes for cleanup

**Review by carlpulley-da — APPROVED** [review]

**Pull request merged**